tools/distpartitioning/data_shuffle.py

```diff
@@ -12,82 +12,109 @@
+                    augment_edge_data, get_ntypes_map, \
                     write_dgl_objects, write_metadata_json
 from gloo_wrapper import alltoall_cpu_object_lst, alltoallv_cpu, \
                     alltoall_cpu, allgather_sizes, gather_metadata_json
 from globalids import assign_shuffle_global_nids_nodes, \
                     assign_shuffle_global_nids_edges, get_shuffle_global_nids_edges
-from convert_partition import create_dgl_object, create_metadata_json, validateDGLObjects
+from convert_partition import create_dgl_object, create_metadata_json
 
-def exchange_node_data(rank, world_size, node_data):
-    """
-    Exchange node_data among the processes in the world
-    Prepare the list of slices targeting each of the process and
-    trigger alltoallv_cpu for the message exchange.
+def gen_node_data(rank, world_size, node_part_ids, ntid_ntype_map, nid_schema_map):
+    '''
+    For this data processing pipeline, reading node files is not needed. All the needed information about
+    the nodes can be found in the metadata json file. This function generates the nodes owned by a given
+    process, using metis partitions. 
 
-    Parameters:
+    Parameters: 
     -----------
     rank : int
-        rank of the current process
+        rank of the process
     world_size : int
-        total no. of participating processes
-    node_data : dictionary
-        nodes data dictionary with keys as column names and values as
-        columns from the nodes csv file
+        total no. of processes
+    node_part_ids : 
+        numpy array, whose length is same as no. of nodes in the graph. Index in this array is the global_nid
+        and value is the partition-id which owns the node
+    ntid_ntype_map : 
+        a dictionary where keys are node_type ids and values are node_type names
+    nid_schema_map: 
+        a dictionary, which is extracted from the input graph metadata json file for node information, 
+        key is `nid`, which is a string. And value is a dictionary which has information about all the node types. 
+        Please note that, it is assumed that for the input graph files, the nodes of a particular node-type are
+        split into `p` files (because of `p` partitions to be generated). On a similar node, edges of a particular
+        edge-type are split into `p` files as well. 
+        for instance, it may be something like this... 
+
+
+            "nid" : { #m : no. of node types
+                "ntype0-name": {
+                "format": "csv",
+                "data" : [ #list of lists
+                            ["<path>/ntype0-name-0.txt", 0, id_end0], # These are type_nids for the nodes
+                            ["<path>/ntype0-name-1.txt", id_start1, id_end1],
+                            ...,
+                            ["<path>/ntype0-name-<p-1>.txt", id_start<p-1>, id_end<p-1>]
+                        ]
+                },
```

Comment:
does the distributed partition script still needs to read the node metadata files? i thought we discussed this and the partition script doesn't need these files any more.

Reply:
Yes. Agreed. The file names are not needed. But we still need the start_id and end_ids for each entry in the "data"'s value. These start and end ids are used by processes to generate the node ids on each process. Since this same metadata json file is also used, in an indirect way, for ParMETIS as well. So eliminating these nodes files from the metadata json file can be a little tricky one to begin with and needs to be handled very carefully.